packages/job-manager/src/worker/worker.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Message, Queue } from 'azure-services';
+import { JobManagerConfig, ServiceConfiguration, System } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger } from 'logger';
+import * as moment from 'moment';
+import { Batch } from '../batch/batch';
+import { JobTaskState } from '../batch/job-task';
+import { PoolLoadGenerator } from '../batch/pool-load-generator';
+
+@injectable()
+export class Worker {
+    public runOnce: boolean = false;
+
+    private jobId: string;
+    private jobManagerConfig: JobManagerConfig;
+    private restartAfterTime: Date;
+
+    public constructor(
+        @inject(Batch) private readonly batch: Batch,
+        @inject(Queue) private readonly queue: Queue,
+        @inject(PoolLoadGenerator) private readonly poolLoadGenerator: PoolLoadGenerator,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) private readonly logger: Logger,
+    ) {}
+
+    public async run(): Promise<void> {
+        await this.init();
+
+        // tslint:disable-next-line: no-constant-condition
+        while (true) {
+            const poolMetricsInfo = await this.batch.getPoolMetricsInfo();
+            const tasksIncrementCount = this.poolLoadGenerator.getTasksIncrementCount(
+                poolMetricsInfo,
+                this.jobManagerConfig.activeToRunningTasksRatio,
+            );
+
+            let tasksQueuedCount = 0;
+            if (tasksIncrementCount > 0) {
+                const scanMessages = await this.getMessages(tasksIncrementCount);
+                if (scanMessages.length === 0) {
+                    this.logger.logInfo(`The storage queue '${this.queue.scanQueue}' has no message to process.`);
+
+                    break;
+                }
+
+                tasksQueuedCount = await this.addTasksToJob(scanMessages);
+            }
+
```

Comment:
this can be done parallely & then wait on Promise.race at the end.

Reply:
or u can use async library

---
In reply to: [309863436](https://github.com/microsoft/accessibility-insights-service/pull/222#discussion_r309863436) [](ancestors = 309863436)

Reply:
won't be part of this pr